connection_util.go

```diff
@@ -24,20 +24,24 @@ func (sc *snowflakeConn) isClientSessionKeepAliveEnabled() bool {
 }
 
 func (sc *snowflakeConn) startHeartBeat() {
-	if !sc.isClientSessionKeepAliveEnabled() {
+	if sc.cfg != nil && !sc.isClientSessionKeepAliveEnabled() {
```

Comment:
why sc.cfg check here? do you mean sc != nil?

Reply:
@sfc-gh-igarish I updated the PR description. sc.cfg and sc.rest are set to nil in the cleanup function but sc is not.